### PR TITLE
[distributed] Add inductor-ready SNN distributed training stack

### PR DESCRIPTION
**Pull request by fangwei123456**
## Summary
- add an experimental DTensor/FSDP2 distributed training stack for activation-based SNNs
- add an inductor backend for standard multi-step IF/LIF/ParametricLIF neurons and use it throughout the distributed scripts/docs
- add distributed Spikformer support, benchmark scripts, tests, and documentation updates

## Bug fixes included
- make standard neuron inductor graph caches track static attributes like threshold/tau/reset/surrogate config instead of reusing stale compiled graphs
- stop advertising unsupported `hybrid` (DDP + TP) entry points in the benchmark and training scripts
- align the distributed training example output with the documented `global_samples/s` metric

## Validation
- local:
  - `pytest test/activation_based/test_distributed_dtensor.py -q`
  - `pytest test/activation_based/test_spikformer_model.py -q`
  - `pytest test/activation_based/test_compile_compat.py -k 'cache_key_tracks' -q`
- server (CUDA):
  - `pytest test/activation_based/test_compile_compat.py -k 'standard_inductor_backend_matches_torch_backend or cache_key_tracks' -q`
  - `pytest test/activation_based/test_distributed_dtensor.py test/activation_based/test_spikformer_model.py -q`

## Release Notes

* **New Features**
  * Added experimental distributed training for SNNs supporting data parallel, tensor parallel, FSDP2, and combined strategies
  * Added Spikformer transformer-based model with pre-configured variants
  * Added torch.compile inductor backend support for neural node execution in multi-step mode
  * Added distributed training scripts and benchmarks

* **Documentation**
  * Comprehensive tutorials and API documentation for distributed training workflows

### CODE REVIEW

**Comment by fangwei123456** [comment]
Superseded by #667. Closing this duplicate PR so the latest distributed/inductor work continues in a single review thread.
